sdk/go/common/diag/sink_test.go

```diff
@@ -64,3 +66,58 @@ func TestEscape(t *testing.T) {
 	pmiss, smiss := sink.Stringify(Error, Message("", "lots of %v %s %d chars"))
 	assert.Equal(t, "error: lots of %!v(MISSING) %!s(MISSING) %!d(MISSING) chars\n", pmiss+smiss)
 }
+
+func TestDefaultSink_concurrency(t *testing.T) {
+	t.Parallel()
+
+	// Verifies that we can safely log to the sink concurrently.
```

Comment:
Should we have a test that writes a sentence per go routine and check none of the sentences are mangled? Just dots feels like they could just be interleaving chars (I think the fix means their not, but seems a nice test to cover)

Reply:
Sure!
FWIW, this test also failed in its current form with or without race detection against the old code: Buffer.Write isn't thread safe so append operations would panic with out of bounds offsets.
But it makes sense to verify the contents too. I'll add content verification (and use something not dot).